db/db_impl/db_impl_experimental.cc

```diff
@@ -153,6 +155,165 @@ Status DBImpl::PromoteL0(ColumnFamilyHandle* column_family, int target_level) {
 
   return status;
 }
+
+IOStatus DBImpl::UpdateManifestForFilesState(
+    const experimental::UpdateManifestForFilesStateOptions& opts) {
+  IOStatus ios = IOStatus::OK();
+
+  // To avoid IO while holding DB mutex, use the following general approach:
+  // * Use GetLiveFilesStorageInfo to get a snapshot of file metadata.
+  // * Do some IO to determine what metadata (by file number) needs to be
+  // updated, from what to what. Note that some previously live files might
+  // already be gone.
+  // * Lock DB mutex to apply them, even though some may no longer apply and
+  // some new files might have been added. And there are gaps in the locking
+  // that have to be considered.
+
+  std::vector<LiveFileStorageInfo> files;
+  LiveFilesStorageInfoOptions lfsi_opts;
+  lfsi_opts.wal_size_for_flush = UINT64_MAX;
```

Comment:
Use the one in port?

Reply:
In ancient times it clearly served a purpose but I do not know why you would endorse its use now (>= C++11).